js/runtime.js

```diff
@@ -78,38 +82,46 @@
 
     var perFrameHandlers;
     var lastTime;
+    var animationFrameHandler;
 
     function clearPerFrameHandlers() {
         perFrameHandlers = [];
-        lastTime = new Date().valueOf();
+        if (animationFrameHandler){
+            cancelAnimationFrame(animationFrameHandler);
```

Comment:
Make sure we really clear the old animationFrameHandler so it doesn't keep running each time we press "Run" and start a new handler. They were accumulating before.